src/safemode_ui.cpp

```diff
@@ -0,0 +1,740 @@
+        return;
+    }
+
+    if( g->u.name == "" ) {
+        popup( _( "No monsters loaded. Please start a game first." ) );
+        return;
+    }
+
+    //Loop through all monster mtypes
+    for( const auto &mtype : MonsterGenerator::generator().get_all_mtypes() ) {
+        creature_name = mtype.nname();
+        if( wildcard_match( creature_name, temp_rules[row_in].rule ) ) {
+            creature_list.push_back( creature_name );
+        }
+    }
+
+    const int offset_x = 15 + ( ( TERMX > FULL_SCREEN_WIDTH ) ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0 );
+    const int offset_y = 5 + ( ( TERMY > FULL_SCREEN_HEIGHT ) ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 :
+                               0 );
+
+    int start_pos = 0;
+    const int content_height = FULL_SCREEN_HEIGHT - 8;
+    const int content_width = FULL_SCREEN_WIDTH - 30;
+    std::ostringstream temp_stream;
+
+    WINDOW *w_test_rule_border = newwin( content_height + 2, content_width, offset_y, offset_x );
+    WINDOW_PTR w_test_rule_borderptr( w_test_rule_border );
+    WINDOW *w_test_rule_content = newwin( content_height, content_width - 2, 1 + offset_y,
+                                          1 + offset_x );
+    WINDOW_PTR w_test_rule_contentptr( w_test_rule_content );
+
+    draw_border( w_test_rule_border );
+
+    int nmatch = creature_list.size();
+    std::string buf = string_format( ngettext( "%1$d monster matches: %2$s",
+                                     "%1$d monsters match: %2$s",
+                                     nmatch ), nmatch, temp_rules[row_in].rule.c_str() );
+    mvwprintz( w_test_rule_border, 0, content_width / 2 - utf8_width( buf ) / 2, hilite( c_white ),
+               "%s", buf.c_str() );
+
+    mvwprintz( w_test_rule_border, content_height + 1, 1, red_background( c_white ),
+               _( "Lists monsters regardless of their attitude." ) );
+
+    wrefresh( w_test_rule_border );
+
+    int line = 0;
+
+    input_context ctxt( "SAFEMODE_TEST" );
+    ctxt.register_updown();
+    ctxt.register_action( "QUIT" );
+
+    while( true ) {
+        // Clear the lines
+        for( int i = 0; i < content_height; i++ ) {
+            for( int j = 0; j < 79; j++ ) {
+                mvwputch( w_test_rule_content, i, j, c_black, ' ' );
+            }
+        }
+
+        calcStartPos( start_pos, line, content_height, creature_list.size() );
+
+        // display safemode
+        for( int i = start_pos; i < ( int )creature_list.size(); i++ ) {
+            if( i >= start_pos &&
+                i < start_pos + ( ( content_height > ( int )creature_list.size() ) ?
+                                  ( int )creature_list.size() :
+                                  content_height ) ) {
+                nc_color line_color = c_white;
+
+                temp_stream.str( "" );
+                temp_stream << i + 1;
+                mvwprintz( w_test_rule_content, i - start_pos, 0, line_color, "%s", temp_stream.str().c_str() );
```

Comment:
`mvwprintz` can do the formatting for you: `mvwprintz(..., "%d", i + 1)`

(And use `std::to_string` if you need to convert a number to a string.)